service/src/main/kotlin/io/provenance/digitalcurrency/consortium/service/AddressTagService.kt

```diff
@@ -0,0 +1,57 @@
+package io.provenance.digitalcurrency.consortium.service
+
+import com.google.protobuf.ByteString
+import io.provenance.digitalcurrency.consortium.config.BankClientProperties
+import io.provenance.digitalcurrency.consortium.config.logger
+import io.provenance.digitalcurrency.consortium.domain.AddressRegistrationRecord
+import io.provenance.digitalcurrency.consortium.domain.AddressRegistrationStatus
+import io.provenance.digitalcurrency.consortium.extension.toByteArray
+import org.springframework.stereotype.Service
+
+@Service
+class AddressTagService(
+    private val pbcService: PbcService,
+    private val bankClientProperties: BankClientProperties
+) {
+    private val log = logger()
+
+    fun createEvent(addressRegistrationRecord: AddressRegistrationRecord) {
+        val existing =
+            pbcService.getAttributeByTagName(addressRegistrationRecord.address, bankClientProperties.kycTagName)
+        when (existing.isEmpty()) {
+            true -> {
+                try {
+                    pbcService.addAttribute(
+                        address = addressRegistrationRecord.address,
+                        tag = bankClientProperties.kycTagName,
+                        payload = ByteString.copyFrom(addressRegistrationRecord.bankAccountUuid.toByteArray())
+                    ).also {
+                        addressRegistrationRecord.txHash = it.txResponse.txhash
+                        addressRegistrationRecord.status = AddressRegistrationStatus.PENDING_TAG
+                    }
+                } catch (e: Exception) {
+                    log.error("Tag failed; it will retry.", e)
+                }
+            }
+            false -> {
```

Comment:
so hopefully this can't happen because we're validating/registering that upfront?

Reply:
that's right

Reply:
should we set an error status instead of complete if this happens?

Reply:
I don't think so - it is already tagged. Maybe we check the value to see if it's different than the bank account uuid, delete it and re-tag?

Reply:
i guess it doesnt really matter - this can only happen if for some crazy reason someone stood up this app with a different database or database gets wiped for the keys

Reply:
yeah agreed. I'll leave it for now.